parser/dsl/lexer.l

```diff
@@ -41,14 +41,19 @@
 <ss_string_literal>{
     "''"               { string_literal += '\''; }
     \\x[0-9A-Fa-f]{2}  {
-                            std::uint8_t value;
-                            const auto err = std::from_chars(YYText() + 2, YYText() + 4, value, 16);
+                            // The buffer needs to hold 2 characters and a null terminator. The first 2 characters are
+                            // being skipped because \x is thrown away (hence, YYText() + 2).
+                            constexpr auto num_chars = 2;
+                            constexpr auto buffer_len = num_chars + 1;
+                            char buffer[buffer_len]{};
+                            std::memcpy(buffer, YYText() + 2, num_chars);
 
-                            if (err.ec == std::errc::invalid_argument) {
-                                throw yy::parser::syntax_error{loc, fmt::format("invalid hex value: [{}]", YYText())};
-                            }
-                            else if (err.ec == std::errc::result_out_of_range) {
-                                throw yy::parser::syntax_error{loc, fmt::format("hex value cannot be represented: [{}]", YYText())};
+                            // Interpret the hex value using std::sscanf and assign to value. If ret is less than 1,
+                            // this means that the single receiving argument (&value) was not successfully assigned.
+                            std::uint8_t value;
+                            if (const int ret = std::sscanf(buffer, "%2x", &value); ret < 1) {
+                                throw yy::parser::syntax_error{
+                                    loc, fmt::format("failed to assign hex value: [{}]", YYText())};
```

Comment:
Let's restore the error message to what the previous wording to keep both branches in sync.

Reply:
I see the issue. There's only one error case now.

It's fine as is.